roles/pulp/templates/settings.py.j2

```diff
@@ -1,7 +1,20 @@
-CONTENT_ORIGIN="http://{{ ansible_hostname }}:8080"
+CONTENT_ORIGIN="http://{{ ansible_hostname }}:24816"
```

Comment:
AFAIK this should be wherever Apache (httpd) presents the content, so end user clients talk to. They shouldn't talk to the internal endpoint.

Reply:
I don't understand, would you mind expanding a little more?

Reply:
Quoting https://pulpproject.org/pulpcore/docs/admin/reference/settings/#content_origin

> A required string containing the protocol, fqdn, and port where the content app is reachable by users. This is used by pulpcore and various plugins when referring users to the content app. For example if the API should refer users to content at using http to pulp.example.com on port 24816, (the content default port), you would set: https://pulp.example.com:24816.

In our deployment we have httpd in front of Pulp which presents it at on `https://{{ ansible_hostname }}`. In other words, this was always wrong, but I'm noticing it now.

Reply:
@ekohl so we basically have to set it up without port? if we want user to point at the right place? to this `CONTENT_ORIGIN="https://{{ ansible_hostname }}"
` (sorry pinging on this again, i'm not good at this part)

Reply:
That's what we do in puppet-pulpcore, yeah:
https://github.com/theforeman/puppet-pulpcore/blob/master/templates/settings.py.erb#L20

You could also argue, that this sort-of blurs the lines between the roles (the plain, port 443, HTTPS is not available w/o the `httpd` role), and the `pulp` role should have a default for the content origin being `http://{{ ansible_fqdn }}:port`, and then we override this in `deploy.yaml` with `https://{{ ansible_fqdn }}` as that's the value the overall deployment uses

Reply:
So it should be a variable on the role that the playbook sets (because that knows both httpd and pulp)?

Reply:
s/variable/default/, but yeah, that's how I'd do it.